File: chanlib/src/main/java/com/mimireader/chanlib/models/ArchivedChanPost.java
package com.mimireader.chanlib.models;

import androidx.annotation.Nullable;

import com.google.gson.annotations.Expose;

public class ArchivedChanPost extends ChanPost {
    public ArchivedChanPost() {
        this.no = -1;
    }
    public ArchivedChanPost(ChanPost other, @Nullable String mediaLink, @Nullable String thumbLink) {
        super(other);
        this.mediaLink = mediaLink;
        this.thumbLink = thumbLink;
    }

    @Nullable
    @Expose
    public String mediaLink;

    @Nullable
    @Expose
    public String thumbLink;
}

File: chanlib/src/main/java/com/mimireader/chanlib/models/ArchivedChanPost.kt
package com.mimireader.chanlib.models

import com.google.gson.annotations.Expose

class ArchivedChanPost : ChanPost {
    constructor() {
        no = -1
    }

    constructor(other: ChanPost, mediaLink: String?, thumbLink: String?) : super(other) {
        this.mediaLink = mediaLink
        this.thumbLink = thumbLink
    }

    @JvmField
    @Expose
    var mediaLink: String? = null

    @JvmField
    @Expose
    var thumbLink: String? = null
}